lib/Sema/CSBindings.cpp

```diff
@@ -2551,12 +2551,27 @@ TypeVariableBinding::fixForHole(ConstraintSystem &cs) const {
 
 
   if (dstLocator->isLastElement<LocatorPathElt::PackElement>()) {
-    const auto& eltPath = dstLocator->getPath();
+    // const auto& eltPath = dstLocator->getPath();
     // When a hole appears as an element of generic pack params
     // the path is ... -> ApplyArgToParam -> PackElement
-    if (eltPath.size() > 1 && (eltPath.end() - 2)->is<LocatorPathElt::ApplyArgToParam>()) {
-      ConstraintFix *fix = SpecifyContextualTypeForNil::create(cs, dstLocator);
-      return std::make_pair(fix, /*impact=*/(unsigned)10);
+    if (dstLocator->getPath().size() > 1 &&
+      (dstLocator->getPath().end() - 2)->is<LocatorPathElt::ApplyArgToParam>()
+      ) {
+
+      SourceRange range;
+      ConstraintLocator* eltDstLocator = simplifyLocator(cs, dstLocator, range);
+
+      // `nil` appears as an element of generic pack params, let's record a
+      // specify contextual type for nil fix.
+      if (isExpr<NilLiteralExpr>(eltDstLocator->getAnchor())) {
+        ConstraintFix *fix =
+          SpecifyContextualTypeForNil::create(cs, eltDstLocator);
+        return std::make_pair(fix, /*impact=*/(unsigned)10);
+      }
+
+      // The hole in a generic pack parameter can not be inferred
+      ConstraintFix *fix = SpecifyTypeForPlaceholder::create(cs, eltDstLocator);
```

Comment:
I couldn't think of an example that triggers `cannot infer pack element # from context`, could you name some?

Reply:
What I was trying to say in my previous comment is that I think we need to add a new ConstraintFix and a new diagnostic to accompany it and record it in `TypeVariableBinding::fixForHole` when the dstLocator ends with `PackElement`.